source/GameInterface/Serialization/Impl/CraftingTemplateBinaryPackage.cs

```diff
@@ -15,11 +17,21 @@ public CraftingTemplateBinaryPackage(CraftingTemplate obj, BinaryPackageFactory
         }
         public override void Pack()
         {
-            templateId = Object.StringId;
```

Comment:
Revert these changes and in your test do something along the lines of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/taverns-serialization/source/GameInterface.Tests/Serialization/SerializerTests/ItemRosterSerializationTest.cs